packages/nimble-components/src/table-column/icon/models/table-column-icon-validator.ts

```diff
@@ -23,22 +24,35 @@ export class TableColumnIconValidator extends TableColumnEnumBaseValidator<
         super(columnInternals, iconValidityFlagNames, [
             MappingIcon,
             MappingSpinner
-        ]);
+        ] as const);
+    }
+
+    private static isSupportedMappingElement(
+        mapping: Mapping<unknown>
+    ): mapping is MappingIcon {
+        return mapping instanceof MappingIcon;
     }
 
     public override validate(
-        mappings: Mapping[],
+        mappings: Mapping<unknown>[],
         keyType: MappingKeyType
     ): void {
         super.validate(mappings, keyType);
         this.validateIconNames(mappings);
+        this.validateNoMissingText(mappings);
     }
 
-    private validateIconNames(mappings: Mapping[]): void {
-        const isMappingIcon = (mapping: Mapping): mapping is MappingIcon => mapping instanceof MappingIcon;
+    private validateIconNames(mappings: Mapping<unknown>[]): void {
         const invalid = mappings
-            .filter(isMappingIcon)
+            .filter(TableColumnIconValidator.isSupportedMappingElement)
             .some(mappingIcon => mappingIcon.resolvedIcon === undefined);
         this.setConditionValue('invalidIconName', invalid);
     }
+
+    private validateNoMissingText(mappings: Mapping<unknown>[]): void {
+        const invalid = mappings
+            .filter(TableColumnIconValidator.isSupportedMappingElement)
```

Comment:
This is a bug, icon columns support IconMapping and SpinnerMapping

Doesn't look like spinner mappings have any tests unfortunately. This was about to introduce a bug that spinner mapping text are no longer validated. Adding some tests.

Reply:
What is the bug? And what tests are missing for spinner mappings? They have just the usual element construction and exported tag tests.

Reply:
There is no test validating spinners have text

Reply:
And the bug is in my change, which misses validating that spinners have text because it only looks at icons.

Reply:
Before it was pushed. It was a note for me to address. I pushed the change that now checks spinners but need tests

Reply:
Added tests